app/models/get_information_about_schools.rb

```diff
@@ -1,13 +1,42 @@
 require 'open-uri'
 require 'csv'
+require 'net/http'
 
 class GetInformationAboutSchools
   URL = 'https://ea-edubase-api-prod.azurewebsites.net/edubase/downloads/public/allgroupsdata.csv'.freeze
 
+  SCHOOLS_URL = 'https://ea-edubase-api-prod.azurewebsites.net/edubase/downloads/public/edubasealldata%{date}.csv'.freeze
+
   def self.trusts_entries
     gias_csv = URI.parse(URL).read.force_encoding(Encoding::ISO8859_1)
     CSV.parse(gias_csv, headers: true).select { |row|
       row['Group Type'].in?(['Single-academy trust', 'Multi-academy trust']) && row['Group Status'] == 'Open'
     }.map(&:to_h)
   end
+
+  def self.schools(&block)
+    file = Tempfile.new
+    fetch_latest_edubase_file(file)
+    SchoolDataFile.new(file.path).schools(&block)
+  ensure
+    file.close
+    file.unlink
+  end
+
+  def self.fetch_latest_edubase_file(file)
+    url = gias_url
+    Net::HTTP.start(url.host, url.port,
```

Comment:
I benchmarked the way it was being used for groups for the schools and I was quite surprised that it was showing around 1GB of memory usage to read in a parse the CSV with that 58MB file. Using the chunked data and `CSV.foreach` brought the memory usage down to between 9 and 30MB I think. Haven't benchmarked since refactoring into its current form but I thought it was better to go with slightly slower processing if favour of being more frugal with memory use.

Reply:
Sounds good - thanks for the explanation.